manta/src/main/java/ch/cyberduck/core/manta/MantaBufferWriteFeature.java

```diff
@@ -0,0 +1,67 @@
+package ch.cyberduck.core.manta;
+
+/*
+ * Copyright (c) 2002-2017 iterate GmbH. All rights reserved.
+ * https://cyberduck.io/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+import ch.cyberduck.core.ConnectionCallback;
+import ch.cyberduck.core.Path;
+import ch.cyberduck.core.exception.BackgroundException;
+import ch.cyberduck.core.features.AttributesFinder;
+import ch.cyberduck.core.features.Find;
+import ch.cyberduck.core.features.MultipartWrite;
+import ch.cyberduck.core.http.HttpResponseOutputStream;
+import ch.cyberduck.core.io.Buffer;
+import ch.cyberduck.core.io.BufferInputStream;
+import ch.cyberduck.core.io.FileBufferSegmentingOutputStream;
+import ch.cyberduck.core.transfer.TransferStatus;
+
+import org.apache.commons.io.IOUtils;
+
+import java.io.IOException;
+
+/**
+ * TODO: not sure how what triggers usage of this class
```

Comment:
The implementation of this interface is not required, when the protocol does not have a multipart upload requirement for large files. This interface is only used in Mountain Duck, where we require a write feature with no prior knowledge of the final content length to be transferred.